src/BigDecimal.ts

```diff
@@ -0,0 +1,717 @@
+/**
+ * This module provides utility functions and type class instances for working with the `BigDecimal` type in TypeScript.
+ * It includes functions for basic arithmetic operations, as well as type class instances for `Equivalence` and `Order`.
+ *
+ * @since 2.0.0
+ */
+
+import * as BigI from "./BigInt"
+import * as Equal from "./Equal"
+import * as equivalence from "./Equivalence"
+import { dual } from "./Function"
+import * as Hash from "./Hash"
+import { type Inspectable, NodeInspectSymbol, toString } from "./Inspectable"
+import * as Option from "./Option"
+import * as order from "./Order"
+import type { Ordering } from "./Ordering"
+import { type Pipeable, pipeArguments } from "./Pipeable"
+
+const TypeId: unique symbol = Symbol.for("effect/BigDecimal")
+
+/**
+ * @since 2.0.0
+ * @category symbol
+ */
+export type TypeId = typeof TypeId
+
+/**
+ * @since 2.0.0
+ * @category models
+ */
+export interface BigDecimal extends Equal.Equal, Pipeable, Inspectable {
+  readonly [TypeId]: TypeId
+  readonly value: bigint
+  readonly scale: number
+}
+
+const BigDecimalProto: Omit<BigDecimal, "value" | "scale"> = {
+  [TypeId]: TypeId,
+  [Hash.symbol](this: BigDecimal): number {
+    return Hash.structure({ value: this.value, scale: this.scale })
+  },
+  [Equal.symbol](this: BigDecimal, that: unknown): boolean {
+    return isBigDecimal(that) && equals(this, that)
+  },
+  toString(this: BigDecimal) {
+    return toString(this.toJSON())
+  },
+  toJSON(this: BigDecimal) {
+    return { _id: "BigDecimal", value: this.value, scale: this.scale }
+  },
+  [NodeInspectSymbol]() {
+    return this.toJSON()
+  },
+  pipe() {
+    return pipeArguments(this, arguments)
+  }
+} as const
+
+/**
+ * Checks if a given value is a `BigDecimal`.
+ *
+ * @param u - The value to check.
+ *
+ * @since 2.0.0
+ * @category guards
+ */
+export const isBigDecimal = (u: unknown): u is BigDecimal => typeof u === "object" && u !== null && TypeId in u
+
+/**
+ * Creates a `BigDecimal` from a `bigint` value and a scale.
+ *
+ * @param value - The `bigint` value to create a `BigDecimal` from.
+ * @param scale - The scale of the `BigDecimal`.
+ *
+ * @since 2.0.0
+ * @category constructors
+ */
+export const scaled = (value: bigint, scale: number): BigDecimal => {
+  const o = Object.create(BigDecimalProto)
+  o.value = value
+  o.scale = scale
+  return o
+}
+
+/**
+ * Creates a `BigDecimal` from a `bigint` or `number` value.
+ *
+ * @param value - The `bigint` or `number` value to create a `BigDecimal` from.
+ *
+ * @since 2.0.0
+ * @category constructors
+ */
+export const make = (value: bigint | number): BigDecimal => {
+  if (typeof value === "number") {
+    const [lead, trail = ""] = `${value}`.split(".")
+    return scaled(BigInt(`${lead}${trail}`), trail.length)
+  }
+
+  return scaled(value, 0)
+}
+
+const zero = make(0n)
```

Comment:
If you use the `0n` syntax, the module won't work with Hermes for React Native (unless they have added support recently)

Reply:
this seems to be fixed in Hermes